content/docs/gke/deploy/deploy-cli.md

```diff
@@ -0,0 +1,192 @@
+   
+   ```
+   gcloud auth application-default login
+   ```
+
+1. Create environment variables for your access control services:
+
+    ```bash
+    # If using IAP, create environment variables from the OAuth client ID and
+    # secret that you obtained earlier:
+    export CLIENT_ID=<CLIENT_ID from OAuth page>
+    export CLIENT_SECRET=<CLIENT_SECRET from OAuth page>
+
+    # If using basic auth, create environment variables for username/password:
+    export KUBEFLOW_USERNAME=<your username>
+    export KUBEFLOW_PASSWORD=<your password>
+    ```
+
+1. Download a kfctl release from the [GitHub releases](https://github.com/kubeflow/kubeflow/releases/)
+
+1. Unpack the tar ball
+
+   ```
+    tar -xvf kfctl_${TAG}_${PLATFORM}.tar.gz
+   ```
+
+    * For subsequent steps either add the binary kfctl to your path or else use the full path to the kfctl binary.
+
+1. Run the following commands to set up and deploy Kubeflow:
+
+    ```bash
+    # This is optional - to make kfctl bin easier to use.
+    export PATH=$PATH:${KUBEFLOW_SRC}/bootstrap/bin
+
+    export KFAPP=<YOUR CHOICE OF APPLICATION DIRECTORY NAME>
+    # Default will be using IAP.
+    kfctl init ${KFAPP} --platform gcp --project ${PROJECT}
+    # or if you want to use basic auth:
+    kfctl init ${KFAPP} --platform gcp --project ${PROJECT} --use_basic_auth -V
+
+    cd ${KFAPP}
+    kfctl generate all -V
+    kfctl apply all -V
+    ```
+   * **${KFAPP}** - the _name_ of a directory where you want kubeflow 
+     configurations to be stored. This directory is created when you run init.
+     The value of this variable cannot be greater than 25 characters. It must
+     contain just the directory name, not the full path to the directory.
+     The value of this variable becomes the name of your deployment.
+     The contents of this directory are described in the next section.
+   * **${PROJECT}** - the _name_ of the GCP project where you want kubeflow deployed to.
+   * During `kfctl init` you need to choose to use either IAP or basic auth.
+   * `kfctl generate all` will try to fetch user email address from your 
+     credential. If it can't find a valid email address, you need to pass a
+     valid email address with flag `--email <your email address>`. This email 
+     address will be configured to be an admin of your kubeflow deployment.
+
+1. Check the resources deployed in namespace `kubeflow`:
+
+    ```
+    kubectl -n kubeflow get  all
+    ```
+1. Storage will be a separete deployment. After `kfctl apply` you should notice
+   there will be 2 deployments(clusters):
+   * **{KFAPP}-storage**: This deployment has persistent volumes for your
+     pipelines.
+   * **{KFAPP}**: This deployment has all the components Kubeflow provides.
+
+1. Kubeflow will be available at the following URI:
+
+    ```
+    https://<deployment_name>.endpoints.<project>.cloud.goog/
+    ```
+   * It can take 10-15 minutes for the URI to become available.
```

Comment:
Last time I counted time to ready is ~20 minutes for IAP, is it becoming longer than before?

Reply:
Updated.